api/envoy/extensions/access_loggers/fluentd/v3/fluentd.proto

```diff
@@ -22,8 +22,20 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // the Fluentd Forward Protocol as described in: `Fluentd Forward Protocol Specification
 // <https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1>`_.
 // [#extension: envoy.access_loggers.fluentd]
-// [#next-free-field: 7]
+// [#next-free-field: 8]
 message FluentdAccessLogConfig {
+  message RetryOptions {
+    // The number of times the logger will attempt to connect to the upstream.
+    // The default value is 1, i.e. only a single connection attempt will be made and no reconnection
+    // attempts will be made if the initial connection had failed.
+    // Note: Reconnection attempts will also be made if the initial connection was established, and
+    // disconnected after some time.
+    // [#comment:TODO(ohadvano): add option to reconnect only if the initial handshake failed.]
+    // Caution: while attempting to reconnect, the log buffer may continue to fill. It is recommended
+    // to set this field to a reasonable number to avoid too many log records buffered.
```

Comment:
Seems like there should be a max buffer setting anyway? Can we add a TODO also about that?

Reply:
There is max buffer option, which sets ad-hoc flush trigger, meaning that if the buffer size passes that value, the buffer is flushed immedietely. In the upstream is disconnected, the buffer is cleared and new logs are lost